core/modules/czar/python/appInterface.py

```diff
@@ -83,17 +83,34 @@ def _maybeCallWithThread(self, function):
             return False
         pass
 
-    def submitQuery(self, query, conditions):
-        """Issue a query.  Params: query, conditions.
-        @returns (result table name, lock/message table name, error)
-
-        Does not block for query completion."""
+    def submitQuery(self, userquery, conditions):
+        """
+        Entry point for query processing
+
+        Parse the query and prepare its execution context (i.e. result/message table, parallel and merge
+        queries), launch a thread for query execution (does not block for query completion as mysql-proxy is
+        mono-threaded), and returns to mysql-proxy.
+
+        @param userquery:   SQL Query sent by mysql-proxy
+        @param conditions:
+        @return             A list of value for mysql-proxy (result, message, orderby, error) where:
+                            - result: name of MySQL table containing query result for proxy
+                            - message: name of  MySQL table containing error/log messages, this table will
+                              be locked during query execution, and mysql-proxy will wait for the lock to be
```

Comment:
What mysql-proxy does should probably not be documented here. Just describe what czar is doing, we don't want to care here about one particular client.